pubsub/gcp/gcp.go

```diff
@@ -142,10 +137,25 @@ func NewPublisher(ctx context.Context, cfg Config, opts ...option.ClientOption)
 	if err != nil {
 		return nil, err
 	}
-
-	return &publisher{
-		topic: c.Topic(cfg.Topic),
-	}, nil
+	t := c.Topic(cfg.Topic)
+	// Update PublishSettings from cfg.PublishSettings
+	// Never set thresholds lower than the defaults
```

Comment:
The PublishSettings will trigger sending the batch based on the first threshold hit. If the client setting up the config sets only one value in the config PublishSettings, the others values will be initialized to 0 and render the desired threshold moot. These checks are intended to guard against that.

Reply:
This has the unfortunate side effect that you can't freely set thresholds to lower than the default values. Are these defaults hard minimums? Can the values be lower than the default? 
If the can be lower then I think we shouldn't impose this limit.
What about checking if the passed `PublishSettings` values is > 0 ?

Reply:
I went back and forth on this. I was attempting to avoid issues with people setting absurdly low values, but if google lets them do it, I suppose we should too. The > 0 route does prevent the 0 init'd values.  I'll update.